### PR TITLE
ENG-6488 - Flush queue on stop, validate response to correctly catch errors

### PR DESCRIPTION
**Pull request by kevinsites**

### CODE REVIEW

**Review by quacke-nid — APPROVED** [review]

**Review by priya-xavier — APPROVED** [review]

File: NeuroID/NeuroIDClass/NeuroID.swift [hunk 1]
```diff
@@ -104,21 +103,21 @@ public enum NeuroID {
...
     public static func stop() {
         NIDPrintLog("NeuroID Stopped")
-        setUserDefaultKey(Constants.storageLocalNIDStopAllKey.rawValue, value: true)
         do {
             _ = try closeSession(skipStop: true)
         } catch {
             NIDPrintLog("NeuroID Error: Failed to Stop because \(error)")
         }
 
+        NeuroID.groupAndPOST()
```
Comment (kishisaka-nid) [inline NeuroID/NeuroIDClass/NeuroID.swift:112]: Does this do event flush?

Comment (kevinsites) [reply NeuroID/NeuroIDClass/NeuroID.swift:112]: Correct, that is the function that sends all the events

**Comment by kishisaka-nid** [comment]
Got 4 unit tests failing here: 
Failing tests:
	NIDSessionTests.test_closeSession()
	NeuroIDClassTests.test_isStopped()
	NeuroIDClassTests.test_start()
	NeuroIDClassTests.test_stop()

**Review by kishisaka-nid — APPROVED** [review]
LGTM! 🚢 it!

**Review by priya-xavier — APPROVED** [review]

**Review by quacke-nid — APPROVED** [review]

**Pull request merged**